tests/cpp/parser_tests.cpp

```diff
@@ -2428,22 +2428,65 @@ rule math_module {
 }
 
 TEST_F(ParserTests,
-PeModuleWorks) {
+PeModuleWorks1) {
 	prepareInput(
 R"(
 import "pe"
 
 rule pe_module {
 	condition:
-		pe.exports("ExitProcess") and pe.characteristics & pe.DLL
+		pe.version_info["CompanyName"] == "company"
+}
+)");
+
+	EXPECT_TRUE(driver.parse());
+
+	ASSERT_EQ(1u, driver.getParsedFile().getRules().size());
+
+	const auto& rule = driver.getParsedFile().getRules()[0];
+	Expression::Ptr condition = rule->getCondition();
+	EXPECT_TRUE(condition->isBool());
+	auto expEq = std::static_pointer_cast<const EqExpression>(condition);
+	EXPECT_EQ(R"(pe.version_info["CompanyName"] == "company")", expEq->getText());
+
+	auto expLeft = std::static_pointer_cast<const ArrayAccessExpression>(expEq->getLeftOperand());
+	EXPECT_EQ(R"(pe.version_info["CompanyName"])", expLeft->getText());
+	auto expItem = expLeft->getAccessor();
+	EXPECT_EQ(R"("CompanyName")", expItem->getText());
+	auto arraySymbol = expLeft->getSymbol();
+	EXPECT_NE(arraySymbol, nullptr);
```

Comment:
Before this test, we didn't check if the constructed expression tree has proper structure. We only tested the getText method.